client/normal_scalar.go

```diff
@@ -87,13 +94,17 @@ func (v normalDocument) Document() (*Document, bool) {
 }
 
 type normalJSON struct {
```

Comment:
question: is there a reason why we have normal json (non-optional) and don't have normal nillable json?

Reply:
Since JSON has a `null` value I didn't think it made sense to also represent that within our type system. It would also cause a bit of confusion if `null` was valid for both `NillableJSON` as well as `JSON`. This also applies to arrays of JSON.